abra_core/std/prelude.abra

```diff
@@ -55,7 +55,8 @@ type RangeIterator {
   }
 }
 
-func range(start: Int, end: Int, stepBy = 1): RangeIterator = RangeIterator(start: start, end: end, stepBy: stepBy)
+func range(start: Int, end: Int): RangeIterator = RangeIterator(start: start, end: end, stepBy: 1, _i: 0)
+func rangeBy(start: Int, end: Int, stepBy: Int): RangeIterator = RangeIterator(start: start, end: end, stepBy: stepBy, _i: 0)
```

Comment:
TODO: remove this. This is a temporary function to support the use-case (only used in tests) of passing a `stepBy` parameter value. The selfhosted compiler does not yet support calling functions with optional parameters, so in order to fulfill the goal of successfully doing codegen for `Array#toString`, I needed to add this.